src/main/kotlin/com/cosmotech/api/azure/containerregistry/AzureContainerRegistryClient.kt

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.azure.containerregistry
+
+import com.azure.containers.containerregistry.ContainerRegistryClient
+import com.azure.containers.containerregistry.ContainerRegistryClientBuilder
+import com.azure.containers.containerregistry.models.ContainerRegistryAudience
+import com.azure.core.credential.TokenCredential
+import com.azure.identity.ClientSecretCredentialBuilder
+import com.cosmotech.api.config.CsmPlatformProperties
+import com.cosmotech.api.exceptions.CsmResourceNotFoundException
+import java.lang.reflect.InvocationTargetException
+import java.util.stream.Collectors
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Service
+
+private const val REGISTRY_NAME = "csmenginesdev"
+
+@Service("csmContainerRegistry")
+@ConditionalOnProperty(name = ["csm.platform.vendor"], havingValue = "azure", matchIfMissing = true)
+class AzureContainerRegistryClient(private val csmPlatformProperties: CsmPlatformProperties) {
+
+  private val logger = LoggerFactory.getLogger(AzureContainerRegistryClient::class.java)
+
+  private fun getEndPoint() = "https://" + REGISTRY_NAME + ".azurecr.io"
```

Comment:
Should be configurable and defined in `csmPlatformProperties`. Every platform has their own registry